apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -23,13 +30,15 @@ export interface AichatState {
   showWarningModal: boolean;
   // Denotes if there is an error with the chat completion response
   chatMessageError: boolean;
+  levelAiCustomizations: LevelAiCustomizations;
```

Comment:
I think these can just be `AiCustomizations`. The other properties in `LevelAiCustomizations` (visibility and `hidePresentationPanel`) are effectively read only values that we can just grab from the level properties in lab2 redux when required. That hopefully might make the reducer a little simpler?

Also minor, but maybe we can call this something like `currentAiCustomizations` to reflect that it's the current state of the data in the student's customizations tabs (as opposed to what was provided by the level)?

Reply:
Yeah I was torn on this...I thought it was cleaner to have all of the reading from redux within each component happen from `aichatRedux`, rather than have to read from lab2 redux to get visibility for each property, then from `aichatRedux` to get/set values. You think better to read from both places in redux in the components?

Good call re: currentAiCustomizations, will update.

Reply:
Yeah, I think it's fine either way - agree that it would be nicer to read it from one store, but also I do think keeping them separate enforces a separation between readonly, level-specific data and mutable properties that are being actively updated by the user. Also I do think there's value in decoupling these two pieces, since it gives us a little bit more flexibility in the structure of the data as it applies to the student-facing UI vs the structure of the data that is actually saved to the project/read from the level. For now they're the same (and will probably stay the same) but hypothetically if there was some text field that was only added by the student and not read from the level or the project - that would stay in aichat redux along with these properties and wouldn't have to be added to/reconciled with the `LevelAiCustomizations` model.

Reply:
Ok, that sounds reasonable. Let me take a pass that uses both redux stores rather than just aichat and see how it looks.